### PR TITLE
added network plots for annotation nomenclature

### PR DESCRIPTION
**Pull request by nahanoo**
I added two network plots which visualize the connection of the nomenclature defined in `besca/datasets/genesets/CellNames_scseqCMs6_config.tsv` and `besca/datasets/genesets/CellNames_scseqCMs6_config.mouse.tsv`.

### CODE REVIEW

File: besca/pl/_nomenclature_network.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from matplotlib import pyplot as plt
+import pandas as pd
+_nomenclature_network.import importlib
+
+networkx_import = importlib.util.find_spec('networkx')
```
Comment (ajulienla) [inline besca/pl/_nomenclature_network.py:5]: Would this create an issue when loading besca if networkx or pydot is absent? 
I would have assume it should be tried when nomenclature_nework is used.

File: besca/pl/_nomenclature_network.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+pydot_import = importlib.util.find_spec('pydot')
+
+if (networkx_import or pydot_import) is None:
+    raise ImportError(
+        "_nomenclature_network.py requires networkx and pydot. Install with pip install pydot and pip install networkx")
+else:
+    import networkx as nx
```
Comment (ajulienla) [inline besca/pl/_nomenclature_network.py:12]: Is Import pydot missing ?

File: besca/pl/_nomenclature_network.py (hunk 1 above)
Comment (ajulienla) [reply besca/pl/_nomenclature_network.py:5]: (ie. Installing Besca would work but importing besca would fail)

Comment (nahanoo) [reply besca/pl/_nomenclature_network.py]: This does not create an issue beacause I found out that `networkx` is actually installed by another besca dependency.

File: besca/pl/_nomenclature_network.py (hunk 2 above)
Comment (nahanoo) [reply besca/pl/_nomenclature_network.py]: `pydot` is imported with `networkx` but is not listed as a dependency of `networkx`. Thus it is not automatically installed with `networkx`.

**Comment by nahanoo** [comment]
Alright I did a lot of testing and I think the current state is a convenient solution.

`import networkx` does not break the package because it is installed by another `besca` dependency.
For positioning the network I use a function `nx_pydot.pydot_layout` which is based on `pydot`. `pydot` is unfortunately not listed as a requirement of `networkx` and not therefore not installed. I check for `pydot` and raise an `ImporError` with installation instructions if not found.

**Comment by ajulienla** [comment]
Generating function added with follow-up for the documentation.
All seem good.

**Pull request merged**